server/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java

```diff
@@ -90,10 +88,10 @@ protected void documentation(StringBuilder sb) {
     public RestChannelConsumer doCatRequest(final RestRequest request, final NodeClient client) {
         final String[] indices = Strings.splitStringByCommaToArray(request.param("index"));
         final ClusterStateRequest clusterStateRequest = new ClusterStateRequest();
-        if (request.hasParam("local")) {
-            DEPRECATION_LOGGER.deprecate("local", LOCAL_DEPRECATED_MESSAGE);
+        // needed only in v8 to catch breaking usages and can be removed in v9
+        if (request.hasParam("local")  && Version.CURRENT.major == Version.V_7_0_0.major + 1) {
+            throw new IllegalArgumentException("parameter [local] is not supported");
```

Comment:
You don't need to add this I believe, we automatically throw exceptions on unrecognized parameters:

```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "request [/_cat/shards] contains unrecognized parameter: [aoeu]"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "request [/_cat/shards] contains unrecognized parameter: [aoeu]"
  },
  "status" : 400
}
```